hydroflow/tests/lattice_join.rs

```diff
@@ -0,0 +1,134 @@
+use hydroflow::util::collect_ready;
+use hydroflow_macro::hydroflow_syntax;
+use multiplatform_test::multiplatform_test;
+
+#[multiplatform_test]
+pub fn test_lattice_join_reducing_behavior() {
+    // lattice_fold has this particular behavior where it uses LatticeFrom to convert the first element into the another lattice type and then it folds into that accumulator.
+    // This is to support things like SetUnionSingletonSet -> SetUnionHashSet. It also helps with things like Min, where there is not obvious default value so you want reduce-like behavior.
+    // There's kind of 3 ways you can do this reducing thing, you can do Fold(default_fn, fold_fn), Reduce(reduce_fn), and the third way which is halfway between the two previous approaches, FoldFrom(From_fn, Fold_fn).
+    // The characteristics of the three are, Fold requires a default function, Reduce requires Fn(&mut X, X) -> X, and FoldFrom requires a converter function From_fn: Fn(X) -> A, Fold_fn: Fn(&mut A, X)
+    let mut df = hydroflow_syntax! {
+        use lattices::Min;
+        use lattices::Max;
+
+        source_iter([(7, Min::new(5)), (7, Min::new(6))]) -> [0]my_join;
+        source_iter([(7, Max::new(5)), (7, Max::new(6))]) -> [1]my_join;
+
+        my_join = lattice_join::<Min<usize>, Max<usize>>()
+            -> assert_eq([(7, (Min::new(5), Max::new(6)))]); // If Min::default is used, then this would likely produce (7, (0, 6)), which is incorrect.
+    };
```

Comment:
Is the comment about `::default` still relevant after this morning's discussion?

Reply:
I think so, this test ensures lattice_join won't regress on this particular bug involving default. If sometime in the far future we decide to re-implement default for min and we implement it incorrectly then this test will fail and force us to reconsider what we're doing.